Assets/GameAssets/Player/PlayerScripts/PlayerInteraction.cs

```diff
@@ -0,0 +1,108 @@
+/**********************************************************************************************************************
+// File Name :         PlayerInteraction.cs
+// Author :            Alex Kalscheur
+// Creation Date :     11/10/24
+// 
+// Brief Description : Controls the interaction system on the player side
+**********************************************************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+[RequireComponent(typeof(PlayerInput))]
+
+/// <summary>
+/// Controls the interaction system on the player side
+/// </summary>
+public class PlayerInteraction : MonoBehaviour
+{
+    [Tooltip("How far can the player reach interactable objects")]
+    [SerializeField] private float _maxReach;
+    [Tooltip("Ray that is cast from camera to find interactable objects")]
+    Ray ray;
+
+    //Input
+    private PlayerInput _playerInput;
+    private InputAction _interactInput;
+
+
+    /// <summary>
+    /// Every frame, Ray is cast.  If the ray hits something and the button is
+    /// </summary>
+    private void Update()
```

Comment:
I am a coroutine enjoyer and so I would prefer it that way. Not a requirement

Reply:
Im fine with update being used here, although I do agree it should be in it's own function